app/models/chargeback/consumption_with_rollups.rb

```diff
@@ -0,0 +1,44 @@
+class Chargeback
+  class ConsumptionWithRollups < Consumption
+    delegate :timestamp, :resource, :resource_id, :resource_name, :resource_type, :parent_ems,
+             :hash_features_affecting_rate, :tag_list_with_prefix, :parents_determining_rate,
+             :to => :first_metric_rollup_record
+
+    def initialize(metric_rollup_records, start_time, end_time)
+      super(start_time, end_time)
+      @rollups = metric_rollup_records
+    end
+
+    def tag_names
+      first_metric_rollup_record.tag_names.split('|')
+    end
+
+    def max(metric)
+      values(metric).max
```

Comment:
also of note, this is not new code, it is moved code. so we need to take that into account for this review